static/js/lib/validation/profile.js

```diff
@@ -105,13 +108,26 @@ export const personalValidation = (profile: Profile) => {
     delete errors.postal_code;
   }
 
+  if (profile.first_name && profile.first_name.length > 30) {
+    errors.first_name = "Given name must be less than 30 characters";
+  }
+  if (profile.last_name && profile.last_name.length > 50) {
+    errors.last_name = "Family name must be less than 50 characters";
+  }
+
   if (shouldRenderRomanizedFields(profile)) {
     if (!profile.romanized_first_name || !CP1252_REGEX.test(profile.romanized_first_name)) {
-      errors.romanized_first_name = "Latin first name is required";
+      errors.romanized_first_name = "Latin given name is required";
+    }
+    else if (profile.romanized_first_name.length > 30) {
+      errors.romanized_first_name = "Given name must be less than 30 characters";
     }
 
     if (!profile.romanized_last_name || !CP1252_REGEX.test(profile.romanized_last_name)) {
-      errors.romanized_last_name = "Latin last name is required";
+      errors.romanized_last_name = "Latin family name is required";
+    }
+    else if (profile.romanized_last_name.length > 50) {
+      errors.romanized_last_name = "Family name must be less than 50 characters";
```

Comment:
what do you think about abstracting the `CP1252` and length checks? Something like:

```js
const checkCP1252 = (profile, key, label) => (
  CP1252_REGEX.test(_.get(profile, key)) ? { [key]: `${label} must be in Latin characters` : null
);

const lengthCheck = (profile, key, requiredLength, label) => (
  _.get(profile, key).length > requiredLength ? { [key]: `${label} must be less than ${requireLength} characters` } : null
);
```

Just so we reduce a little bit of code duplication. Up to you!